keras/src/callbacks/orbax_checkpoint.py

```diff
@@ -297,3 +359,6 @@ def wait_until_finished(self):
                 import time
 
                 time.sleep(0.1)
+
+        # Multi-host synchronization: ensure all hosts complete
+        self._sync_processes("checkpoint_wait_complete")
```

Comment:
Just curious, if you're doing `_sync_processes`, do you still need the `wait` before?

Reply:
Yes This ensures that the local asynchronous checkpoint operations on the current host have actually completed. Without this, the host might proceed while its save operations are still running in the background._sync_processes this acts as a barrier synchronization across all hosts